slf4j/src/main/java/com/google/common/flogger/backend/slf4j/Slf4jLoggerBackend.java

```diff
@@ -0,0 +1,188 @@
+
+import java.util.logging.Level;
+
+/**
+ * A logging backend that uses slf4j to output log statements.
+ */
+final class Slf4jLoggerBackend extends LoggerBackend implements
+    SimpleMessageFormatter.SimpleLogHandler {
+
+  private final Logger logger;
+
+  Slf4jLoggerBackend(Logger logger) {
+    if (logger == null) {
+      throw new NullPointerException("logger is required");
+    }
+    this.logger = logger;
+  }
+
+  // represents the log levels supported by SLF4J, used internally to dispatch calls accordingly
+  private enum Slf4jLogLevel {
+    TRACE, DEBUG, INFO, WARN, ERROR
+  }
+
+  /**
+   * Adapts the JUL level to SLF4J level per the below mapping:
+   *
+   * <table>
+   * <tr>
+   * <th>JUL</th>
+   * <th>SLF4J</th>
+   * </tr>
+   * <tr>
+   * <td>FINEST</td><td>TRACE</td>
+   * </tr><tr>
+   * <td>FINER</td><td>TRACE</td>
+   * </tr>
+   * <tr>
+   * <td>FINE</td><td>DEBUG</td>
+   * </tr>
+   * <tr>
+   * <td>CONFIG</td><td>DEBUG</td>
+   * </tr>
+   * <tr>
+   * <td>INFO</td>
+   * <td>INFO</td>
+   * </tr>
+   * <tr>
+   * <td>WARNING</td>
+   * <td>WARN</td>
+   * </tr>
+   * <tr>
+   * <td>SEVERE</td>
+   * <td>ERROR</td>
+   * </tr>
+   * </table>
+   *
+   * <p>Custom JUL levels are mapped to the next-lowest standard JUL level; for example, a custom
+   * level at 750 (between INFO:800 and CONFIG:700) would map to the same as CONFIG (DEBUG).</p>
+   *
+   * <p>It isn't expected that the JUL levels 'ALL' or 'OFF' are passed into this method;
+   * doing so will throw an IllegalArgumentException, as those levels are for configuration, not
+   * logging</p>
+   *
+   * @param level the JUL level to map; any standard or custom JUL level, except for ALL or OFF
+   * @return the MappedLevel object representing the SLF4J adapters appropriate for the requested
+   * log level; never null.
+   */
+  private static Slf4jLogLevel mapToSlf4jLogLevel(Level level) {
+    int requestedLevel = level.intValue();
+
+    if (requestedLevel == Level.ALL.intValue() || requestedLevel == Level.OFF.intValue()) {
+      // Flogger doesn't allow ALL or OFF to be used for logging, only for configuration
+      throw new IllegalArgumentException("Unsupported log level: " + level);
+    }
+
+    // FINEST, FINER -> TRACE
+    // custom JUL levels less than FINE -> TRACE
+    if (requestedLevel < Level.FINE.intValue()) {
+      return Slf4jLogLevel.TRACE;
+    }
+
+    // FINE, CONFIG -> DEBUG
+    // custom JUL levels less than INFO -> DEBUG
+    if (requestedLevel < Level.INFO.intValue()) {
+      return Slf4jLogLevel.DEBUG;
+    }
+
+    // INFO -> INFO
+    // custom JUL levels less than WARNING -> INFO
+    if (requestedLevel < Level.WARNING.intValue()) {
+      return Slf4jLogLevel.INFO;
+    }
+
+    // WARNING -> WARN
+    // custom JUL levels less than SEVERE -> WARN
+    if (requestedLevel < Level.SEVERE.intValue()) {
+      return Slf4jLogLevel.WARN;
+    }
+
+    // SEVERE -> ERROR
+    // custom JUL levels greater than SEVERE -> ERROR
```

Comment:
"greater than or equal to" ...
But to be honest, the code is obvious enough that I'd just remove all these comments.
The JavaDoc is clear enough and the code is simple enough that I don't think they're needed (and the custom level thing is mentioned in the JavaDoc, so that's covered).
Completely your call however.